WPMedia/arc-themes-components
# THEMES-1228: Replace @wpmedia/timezone

## Ticket

- [THEMES-1228](https://arcpublishing.atlassian.net/browse/THEMES-1228)

## Description

This PR removes the timezone library from `arc-themes-components`, and keeps support for the dateFormat/dateTimeFormat settings in `blocks.json`.

## Acceptance Criteria

_copy from ticket_

## Test Steps

1. Checkout branch - `git checkout THEMES-1228-date-format-conversion`
2. Update dependencies - `npm i`
3. Run PageBuilder (`npx fusion start`) with `THEME_COMPONENTS_REPO` set in the `.env`.
4. Dates and times on bylines in promo blocks, list blocks, and the date block on articles should all appear as they did before.
5.  Try various settings in the `blocks.json` for the localization prop like so:
```
"dateLocalization": {
  "language": "fr-FR",
  "timeZone": "America/Los_Angeles",
  "dateTimeFormat": "%A %B %d, %Y at %H:%M:%S %Z",
  "dateFormat": "%a %m-%d-%y"
},
```
6. The output should change to match the format, use the language specified, and have the time converted to the specified time zone.
7. Check the `buildDateString()` function for all of the supported tokens. Note that we can only support having one token for each part, as the display option needs to be set before we generate the parts. So a format with all of the month tokens, `%b %B %h %m`, would show `01 01 01 01` as they all rely on the `month` option for `Intl.DateTimeFormat()`.

## Related PRs
- Manifests: https://github.com/WPMedia/arc-themes-manifests/pull/225

## Author Checklist


## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._



[THEMES-1228]: https://arcpublishing.atlassian.net/browse/THEMES-1228?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
